### PR TITLE
Implementing parallel testing

### PR DESCRIPTION
**Pull request by burtonemily**
Adds in fixtures for basic test environments, create two accounts, link device, create contact and close all devices. Started updating user actions tests to include new fixture

### CODE REVIEW

File: run/test/specs/community_tests_image.spec.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+//   await device1.sendMessage(testMessage);
+//   await device1.sendImage(platform, testImageMessage, true);
+//   await device2.replyToMessage(userA, testImageMessage);
+//   closeApp(device1, device2);
+// }
+
+// async function sendImageCommunityAndroid(platform: SupportedPlatformsType) {
+//   const { device1, device2 } = await openAppTwoDevices(platform);
+//   const testCommunityLink = `https://chat.lokinet.dev/testing-all-the-things?public_key=1d7e7f92b1ed3643855c98ecac02fc7274033a3467653f047d6e433540c03f17`;
+//   const testCommunityName = `Testing All The Things!`;
```
Comment (Bilb) [inline run/test/specs/community_tests_image.spec.ts]: this whole file should be kept as skipped test maybe?

File: run/test/specs/disappearing_call.spec.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+// iosIt("Disappearing call message 1o1", disappearingCallMessage1o1Ios);
+// androidIt("Disappearing call message 1o1", disappearingCallMessage1o1Android);
+
+// async function disappearingCallMessage1o1Ios(platform: SupportedPlatformsType) {
+//   const { device1, device2 } = await openAppTwoDevices(platform);
+//   const time: DMTimeOption = DISAPPEARING_TIMES.ONE_MINUTE;
+//   // Create user A and user B
+//   const [userA, userB] = await Promise.all([
+//     newUser(device1, "Alice", platform),
```
Comment (Bilb) [inline run/test/specs/disappearing_call.spec.ts]: the whole file is expected to be commented?

File: run/test/specs/utils/capabilities_android.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { W3CCapabilities } from "@wdio/types/build/Capabilities";
 import { isNil, isString } from "lodash";
 import { CapabilitiesIndexType } from "./capabilities_ios";
 
-const androidAppFullPath = `/Users/emilyburton/Documents/session-android/app/build/outputs/apk/play/debug/session-1.18.2-universal.apk`;
+const androidAppFullPath = `/Users/emilyburton/Downloads/session-android-20240531T000357Z-b544961d2-universal/session-1.18.4-universal.apk`;
```
Comment (Bilb) [inline run/test/specs/utils/capabilities_android.ts]: this shouldn't be commited but using a process.env variable

File: run/test/specs/utils/capabilities_ios.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { AppiumXCUITestCapabilities } from "@wdio/types/build/Capabilities";
 import { W3CCapabilities } from "@wdio/types/build/Capabilities";
-// const iosAppFullPath = `/Users/emilyburton/Desktop/Session.app`;
-const iosAppFullPath = `/Users/emilyburton/Downloads/session-ios-20240618T020520Z-a49a27207-sim 2/Session.app`;
+const iosAppFullPath = `/Users/emilyburton/Downloads/Session 3.app`;
```
Comment (Bilb) [inline run/test/specs/utils/capabilities_ios.ts]: same comment

File: run/test/specs/utils/open_app.ts [hunk 5]
```diff
@@ -59,9 +120,9 @@ export const openAppTwoDevices = async (
     openAppOnPlatform(platform, 1),
   ]);
 
-  function closeAllApps() {
-    // do the thing
-  }
+  // function closeAllApps() {
```
Comment (Bilb) [inline run/test/specs/utils/open_app.ts]: can be removed altogether

File: run/test/specs/voice_calls.spec.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+// import {
+//   SupportedPlatformsType,
+//   closeApp,
+//   openAppTwoDevices,
+// } from "./utils/open_app";
+
+// androidIt("Voice calls", voiceCallAndroid);
+// iosIt("Voice calls", voiceCallIos);
+
+// async function voiceCallIos(platform: SupportedPlatformsType) {
```
Comment (Bilb) [inline run/test/specs/voice_calls.spec.ts]: to keep disabled?

File: run/types/sessionIt.ts [hunk 7]
```diff
@@ -56,3 +58,41 @@ export function bothPlatformsIt(
...
+//   const testName = `${title} ios`;
+
+//   return it(testName, async () => {
+//     console.info(`\n\n==========> Running "${testName}"\n\n`);
+//     await test("ios");
+//   });
+// }
+
+// Define the function to accept the title, a setup data object, and a test function that uses this object.
+export function iosItWithSetup(
```
Comment (Bilb) [inline run/types/sessionIt.ts:75]: should be removed?

File: run/test/specs/utils/open_app.ts [hunk 8]
```diff
@@ -225,21 +287,24 @@ const openiOSApp = async (
...
-  // Logging to check that app path is correct
-  console.log(
-    `iOS App Full Path: ${
-      getIosCapabilities(capabilitiesIndex)["alwaysMatch"]["appium:app"]
-    }`
-  );
+  const workerId = parseInt(process.env.MOCHA_WORKER_ID || "0", 10);
+
+  // Calculate the actual capabilities index for the current worker
+  const actualCapabilitiesIndex = (workerId * 4 +
```
Comment (Bilb) [inline run/test/specs/utils/open_app.ts:295]: yeah that works for now.

File: run/test/specs/utils/open_app.ts [hunk 9]
```diff
@@ -225,21 +287,24 @@ const openiOSApp = async (
 }> => {
   console.warn("openiOSApp");
 
-  // Logging to check that app path is correct
-  console.log(
-    `iOS App Full Path: ${
-      getIosCapabilities(capabilitiesIndex)["alwaysMatch"]["appium:app"]
-    }`
-  );
+  const workerId = parseInt(process.env.MOCHA_WORKER_ID || "0", 10);
```
Comment (Bilb) [inline run/test/specs/utils/open_app.ts]: you might want to check for parseInt return as it returns NaN when it cannot be parsed, and I don't think NaN is fasly

File: run/test/specs/utils/open_app.ts [hunk 10]
```diff
@@ -28,13 +32,70 @@ export type SupportedPlatformsType = "android" | "ios";
...
+  };
+  return {
+    devices: [device1, device2, device3],
+    Alice: userA,
+    Bob: userB,
+    closeApp,
+  };
+};
+
+export const setUp1o1TestEnvironment = async (
```
Comment (Bilb) [inline run/test/specs/utils/open_app.ts:63]: do we still need this?

File: run/test/specs/utils/open_app.ts [hunk 11]
```diff
@@ -28,13 +32,70 @@ export type SupportedPlatformsType = "android" | "ios";
 /* ******************Command to run Appium Server: *************************************
 ./node_modules/.bin/appium server --use-drivers=uiautomator2,xcuitest --port 8110 --use-plugins=execute-driver --allow-cors
 */
+// Basic test environment is 3 devices (device1, device2, device3) and 2 users (userA, userB)
+// Device 1 and 3 are linked devicesby userA
+export const createBasicTestEnvironment = async (
```
Comment (Bilb) [inline run/test/specs/utils/open_app.ts:37]: do we still need this?

File: run/test/specs/community_tests_image.spec.ts (hunk 1 above)
Comment (burtonemily) [reply run/test/specs/community_tests_image.spec.ts]: So i removed the describe blocks and there were the only way to use .skip, i can bring them back in

File: run/test/specs/utils/capabilities_android.ts (hunk 3 above)
Comment (burtonemily) [reply run/test/specs/utils/capabilities_android.ts]: Added an env file to moved both ios and android paths to that

File: run/test/specs/voice_calls.spec.ts (hunk 6 above)
Comment (burtonemily) [reply run/test/specs/voice_calls.spec.ts]: Voice calls are really unstable right now, so keeping it skipped for now

File: run/types/sessionIt.ts (hunk 7 above)
Comment (burtonemily) [reply run/types/sessionIt.ts:75]: I guess if we don't want to go down the path of setup data.. then yes?

File: run/test/specs/utils/open_app.ts (hunk 11 above)
Comment (burtonemily) [reply run/test/specs/utils/open_app.ts:37]: This was the starting idea of fixtures, is that still something we want to do? I know its not completely correct in what you were wanting.. but it was a start?

File: run/test/specs/community_tests_image.spec.ts (hunk 1 above)
Comment (Bilb) [reply run/test/specs/community_tests_image.spec.ts]: ok all good

File: run/types/sessionIt.ts (hunk 7 above)
Comment (Bilb) [reply run/types/sessionIt.ts:75]: ok you can keep it but will need some cleaning up

**Pull request merged**